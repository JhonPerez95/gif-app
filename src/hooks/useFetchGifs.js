import { useState, useEffect } from 'react';
import getGif from '../helpers/getGifs';

import PropTypes from 'prop-types';

export const useFetchGifs = (category) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGif(category).then((imgs) => {
      setTimeout(() => {
        console.log(imgs);
        setstate({
          data: imgs,
          loading: false,
        });
      }, 1000);
    });
  }, [category]);

  return state;
};

useFetchGifs.propTypes = {
  category: PropTypes.string.isRequired,
};
