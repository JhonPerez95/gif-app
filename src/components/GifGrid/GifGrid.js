import React from 'react';
import PropTypes from 'prop-types';

import ItemGrid from '../ItemGrid/ItemGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> {category}</h3>

      {loading && <p className="animate__animated animate__flash"> Loading</p>}
      <div className="card-grid">
        {images.map((item) => (
          <ItemGrid key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
