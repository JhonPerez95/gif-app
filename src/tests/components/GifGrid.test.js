import React from 'react';
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe(' Testing al componente GifGrid', () => {
  test('Que se renderice correctamente el componente <GifGrid/>', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category="Bleach" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se carga images useFetchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/test.jpg',
        title: 'test',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category="Bleach" />);

    expect(wrapper).toMatchSnapshot();
  });
});
