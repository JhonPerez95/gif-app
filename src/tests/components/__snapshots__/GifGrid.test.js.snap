// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Testing al componente GifGrid Debe mostrar items cuando se carga images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Bleach
  </h3>
  <div
    className="card-grid"
  >
    <ItemGrid
      id="abc"
      key="abc"
      title="test"
      url="https://localhost/test.jpg"
    />
  </div>
</Fragment>
`;

exports[` Testing al componente GifGrid Que se renderice correctamente el componente <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Bleach
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
